app/services/checkout_service/internal/db/dynamodb.go

```
@@ -0,0 +1,99 @@
+
+	var cfg aws.Config
+	var err error
+
+	if endpoint != "" {
+		// LocalStack configuration
+		logger.WithFields(logrus.Fields{
+			"endpoint": endpoint,
+			"region":   region,
+		}).Info("Configuring DynamoDB client for LocalStack")
+
+		customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
```

golangci-lint (1.64.8)

SA1019: aws.EndpointResolverWithOptionsFunc is deprecated: The global endpoint resolution interface is deprecated. See deprecation docs on [EndpointResolver]. 

(staticcheck)